File: VTI_Galaxy_BE/src/main/java/demo/services/interfaceClass/EmployeeService.java
package demo.services.interfaceClass;

import demo.modal.dto.EmployeeDto;
import demo.modal.request.EmployeeRequest;

import java.util.List;

public interface EmployeeService {
    // get
    List<EmployeeDto> getAllEmployees();
    EmployeeDto getEmployeeById(int id);
    List<EmployeeDto> getByGalaxyId(int galaxyId);

    // post
    EmployeeDto createEmployee(EmployeeRequest request);


    // update
    EmployeeDto updateEmployee(EmployeeRequest request, int id);

    // delete
    EmployeeDto deleteEmployee(int id);
}

File: VTI_Glaxy_FE/src/services/employeeService.js
import axiosClient from "./axiosClient"

const employeeService = {
   fetchEmployees(){
      return axiosClient.get("/getEmployees")
   },
   fetchEmployeeById(employeeId){
      return axiosClient.get(`/getEmployeeById?employeeId=${employeeId}`)
   },
   fetchEmployeeByGalaxyId(galaxyId){
      return axiosClient.get(`/getEmployeeByGalaxyId?galaxyId=${galaxyId}`)
   },
   createEmployee(employeeRequest){
      return axiosClient.post("/postEmployee", employeeRequest)
   },
   updateEmployee(employeeId, employeeRequest){
      return axiosClient.put(`/putEmployee?employeeId=${employeeId}`, employeeRequest)
   },
   deleteEmployee(employeeId){
      return axiosClient.put(`/deleteEmployee?employeeId=${employeeId}`)
   }
}

export default employeeService